internal/controller/controllers/agentserviceconfig_controller_storage_validation_test.go

```diff
@@ -0,0 +1,510 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/bombsimon/logrusr/v3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	clnt "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("Agent service config controller storage validation", func() {
+	var (
+		ctx        context.Context
+		logger     *logrus.Logger
+		client     clnt.WithWatch
+		watches    []watch.Interface
+		subject    *aiv1beta1.AgentServiceConfig
+		key        clnt.ObjectKey
+		request    ctrl.Request
+		reconciler *AgentServiceConfigReconciler
+	)
+
+	BeforeEach(func() {
```

Comment:
This is a really impressive test setup, can we refactor it out of this particular test so it can be reused in future AgentServiceConfig related tests?

I think it will also make the test clearer and more focused as all this setup isn't really specific to what is being tested

Reply:
@omertuc in the new version of the patch I moved that setup to a `FakeCluster` type inside the `internal/testing` package. Let me know what you think.